app/sender/messageSender.go

```diff
@@ -187,10 +194,19 @@ func SendMessageLoop(to string, message string, group bool, att io.Reader, flags
 			}
 		} else {
 			if group {
-				ts, err = textsecure.SendGroupAttachment(to, message, att, timer)
+				if voiceMessage {
+					ts, err = textsecure.SendGroupVoiceNote(to, message, att, timer)
+				} else {
+					ts, err = textsecure.SendGroupAttachment(to, message, att, timer)
+				}
 			} else {
-				log.Printf("[axolotl] SendMessageLoop sendAttachment")
-				ts, err = textsecure.SendAttachment(to, message, att, timer)
+				if voiceMessage {
+					ts, err = textsecure.SendVoiceNote(to, message, att, timer)
+					log.Printf("[axolotl] SendMessageLoop send voice note")
+				} else {
+					log.Printf("[axolotl] SendMessageLoop sendAttachment")
+					ts, err = textsecure.SendAttachment(to, message, att, timer)
+				}
```

Comment:
That's true, but now we have one loop for all kinds of messages. When we start to refactor this function, i would advice to introduce a message queue with channels. I have this on my todo list since a while but then the rust changes have been more important.